### PR TITLE
pull request

### PR DESCRIPTION
**Pull request by SlowMonk**
conflict and hard coding changed

### CODE REVIEW

File: calculator.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    header_skipped = False
+    with open(answer_file) as fh:
+        for line in fh:
+            if not header_skipped:
+                header_skipped = True
+                continue
+            _, cat, gt1, gt2, ans1, ans2 = line.replace('\n', "").split(',')
+            if cat not in result_dict:
+                result_dict[cat] = dict(TP=0, FN=0, FP=0)
+            #for ans in ans1, ans2:
```
Comment (chrishickey) [inline calculator.py]: remove commented out code

File: calculator.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            if ans1 in [gt1, gt2] and ans2 in [gt1, gt2]:
```
Comment (chrishickey) [inline calculator.py]: sorted([ans1, ans2]) == sorted([gt1, gt2])

File: dataset_statistics.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    channels_sum, channels_squared_sum, num_batches = 0, 0, 0
+    count = 0
+    for data, _ in tqdm(dataloader):
+        try:
+            # Mean over batch, height and width, but not over the channels
+            channels_sum += torch.mean(data, dim=[0, 2, 3])
+            channels_squared_sum += torch.mean(data ** 2, dim=[0, 2, 3])
+            num_batches += 1
+            count +=1
+            if count%100==0:
```
Comment (chrishickey) [inline dataset_statistics.py:27]: remove the try except. If there is an error in the data we should know what it is and why. We should not ignore the error

File: main_supcon.py [hunk 4]
```diff
@@ -431,18 +491,28 @@ def main():
...
 
         if epoch % opt.save_freq == 0:
-            save_file = os.path.join(
-                opt.save_folder, f'ckpt_{opt.method}_pretrained_{opt.pretrained}_{opt.group_num}_epoch_{epoch}.pth')
-            save_model(model, optimizer, opt, epoch, save_file)
+
+            weights_path = os.path.join(
+                opt.save_folder,weights_name)
+            file_name = weights_path.replace('pth','csv')
+            save_model(model, optimizer, opt, epoch, weights_path)
```
Comment (chrishickey) [inline main_supcon.py:483]: where do you use file_name>

File: main_supcon.py [hunk 5]
```diff
@@ -402,10 +437,35 @@ def train_category(train_loader, model, criterion, optimizer, epoch, opt):
...
+        score_model,
+        question_dir,
+        file_name,
+    )
+    var=CALCULATOR[question_number](
+
+        answer_file = file_name,
+    )
+    print('VAR->',var)
+    var = list(var)
```
Comment (chrishickey) [inline main_supcon.py]: what does VAR mean?

File: main_supcon.py [hunk 6]
```diff
@@ -402,10 +437,35 @@ def train_category(train_loader, model, criterion, optimizer, epoch, opt):
 
     return losses.avg
 
+def f1_graph(epoch,question_number,model_name,weights_path,opt,question_dir,file_name,title):
```
Comment (chrishickey) [inline main_supcon.py]: add type hints here. This function definition looks messy and has no documentation

File: main_supcon.py [hunk 7]
```diff
@@ -402,10 +437,35 @@ def train_category(train_loader, model, criterion, optimizer, epoch, opt):
...
+    )
+    var=CALCULATOR[question_number](
+
+        answer_file = file_name,
+    )
+    print('VAR->',var)
+    var = list(var)
+
+    if opt.wandb:wandb.log({'micro_f1'+title: var[0]}, step=epoch)
+    if opt.wandb:wandb.log({'macro_f1'+title: var[1]}, step=epoch)
```
Comment (chrishickey) [inline main_supcon.py]: python style guides says do not put if statements on one line. Change this everywhere https://peps.python.org/pep-0008/

File: question_loader.py [hunk 8]
```diff
@@ -72,18 +74,21 @@ def __getitem__(self, idx):
...
-                self.root, dir_name, im['image_url']
-            )).convert('RGB')
-            # Augment the images twice.
-            sample1, sample2 = self.transform(image)
-            samples1.append(sample1)
-            samples2.append(sample2)
+
+            try:
+                image = Image.open(os.path.join(
+                    self.root, dir_name, im['image_url']
```
Comment (chrishickey) [inline question_loader.py]: remove these except blocks and change back to the original logic. If there are errors here we need to stop training and debug them, not ignore them

File: main_supcon.py [hunk 9]
```diff
@@ -402,10 +437,35 @@ def train_category(train_loader, model, criterion, optimizer, epoch, opt):
...
+
 
 def main():
     opt = parse_option()
-
+    if opt.wandb:wandb.init(project="50percent", entity=opt.wandb_id)
```
Comment (chrishickey) [inline main_supcon.py]: make project name configurable

File: main_supcon.py (hunk 6 above)
Comment (SlowMonk) [reply main_supcon.py]: right now this function not needed its for validation and test fscore during test

**Comment by SlowMonk** [comment]
fix the issues

**Pull request closed without merging**